pkg/yang/modules.go

```diff
@@ -39,7 +39,12 @@ type Modules struct {
 	// ignored. The keys of the map are a string that is formed by concatenating
 	// the name of the including (sub)module and the included submodule.
 	mergedSubmodule map[string]bool
-	mu              sync.Mutex // Mutex to protect byNS map
+	// ParseOptions sets the options for the current YANG module parsing. It can be
+	// directly set by the caller to influence how goyang will behave in the presence
+	// of certain exceptional cases.
+	ParseOptions Options
+	// Mutex to protect byNS map
+	mu sync.Mutex
```

Comment:
It's idiomatic to have the mutex defined above the value that is being protected - so here:

```golang
nsMu sync.Mutex // ...
byNS map[string]*Module // ...
```

I suggest calling it `nsMu` since I'd expect `mu` would protect all of the `map` types in the struct.

Reply:
Done.